File: Python/841 Keys and Rooms.py
# Keys and Rooms
class Solution(object):
    def canVisitAllRooms(self, rooms):
        """
        :type rooms: List[List[int]]
        :rtype: bool
        """
        visited = [False] * len(rooms)
        visited[0] = True
        stack = []
        stack.append(rooms[0])
        while stack:
            keys = stack.pop()
            for key in keys:
                if not visited[key]:
                    visited[key] = True
                    stack.append(rooms[key])
        return sum(visited) == len(rooms)

File: C++/841 Keys and Rooms.cpp
class Solution {
public:
    bool canVisitAllRooms(vector<vector<int>>& rooms) {
        vector<bool> visited(rooms.size(), false);
        visited[0] = true;
        stack<int> mystack;
        mystack.push(0);
        while (!mystack.empty()) {
            int curr = mystack.top();
            mystack.pop();
            for(auto key : rooms[curr]) {
                if (!visited[key]) {
                    visited[key] = true;
                    mystack.push(key);
                }
            }
        }
        for (int i = 0; i < visited.size(); i++) {
            if (visited[i] == false) return false;
        }
        return true;
    }
};